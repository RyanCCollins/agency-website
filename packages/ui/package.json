{
  "name": "openui",
  "version": "1.0.1",
  "description": "An open ui kit made with React and TypeScript",
  "main": "/dist/index.js",
  "repository": "https://github.com/scalable-react/scalable-react-typescript-boilerplate",
  "license": "MIT",
  "scripts": {
    "prepublish": "rimraf dist && tsc"
  },
  "dependencies": {
    "react": "^15.4.2",
    "react-dom": "^15.4.2",
    "react-remarkable": "^1.1.1",
    "recompose": "^0.23.4",
    "rimraf": "^2.6.1",
    "styled-components": "^1.4.4",
    "typescript": "^2.2.1",
    "yarn": "^0.21.3"
  }
}
