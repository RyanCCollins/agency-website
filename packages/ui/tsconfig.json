{
  "compilerOptions": {
    "baseUrl": "src",
    "module": "commonjs",
    "target": "es6",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "noImplicitAny": false,
    "sourceMap": true,
    "outDir": "dist",
    "jsx": "react",
    "noUnusedLocals": true
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules",
    "**/*.test.*",
    "**/*.mock.*"
  ]
}
