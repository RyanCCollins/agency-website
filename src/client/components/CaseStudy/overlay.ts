import styled, { css } from 'styled-components';

function styles() {
  return css`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    transition: background .5s;
    cursor: pointer;
    p {
      margin: 0 !important;
      padding: 0 !important;
      font-family: "Raleway", sans-serif;
      font-weight: 300 !important;
      width: 85%;
      font-size: 2rem !important;
      line-height: 2rem !important;
      color: #fff !important;
      text-align: center;
      -webkit-font-smoothing: antialiased;
      opacity: 0 !important;
      position: absolute;
      top: 50%;
      left: 50%;
      -webkit-transform: translate(-50%, -50%);
      transform: translate(-50%, -50%);
      -webkit-transition: opacity 0.5s !important;
      transition: opacity 0.5s !important;
    }
    &:hover {
      background: rgba(0, 0, 0, 0.8);
      p {
        opacity: 1.0 !important;
      }
    }
  `;
}

export default styled.div`
  ${styles()}
`;
