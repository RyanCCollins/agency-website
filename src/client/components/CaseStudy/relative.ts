import styled, { css } from 'styled-components';

function styles() {
  return css`
    position: relative;
  `;
}

export default styled.div`
  ${styles()}
`;
