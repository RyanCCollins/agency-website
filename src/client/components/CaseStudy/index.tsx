import * as React from 'react';
import { findDOMNode } from 'react-dom';
import { Box, Image, WithAnimation } from 'ui';
import Overlay from './overlay';
import Relative from './relative';

export interface State {
  isVisible: boolean;
}

export interface Props extends React.HTMLProps<typeof CaseStudy> {
  image: string;
  title: string;
}

export default class CaseStudy extends React.Component<Props, State> {
  constructor() {
    super();
    this.handleScroll = this.handleScroll.bind(this);
    this.state = {
      isVisible: false,
    };
  }
  public componentDidMount() {
    if (typeof window !== 'undefined') {
      this.handleScroll();
      window.addEventListener('scroll', this.handleScroll);
    }
  }
  public componentWillUnmount() {
    if (typeof window !== 'undefined') {
      window.removeEventListener('scroll', this.handleScroll);
    }
  }
  public divRef: Element;
  public handleScroll() {
    const windowHeight = window ? window.innerHeight : 1000;
    const node = findDOMNode(this.divRef);
    if (node) {
      const isVisible = node.getBoundingClientRect().top < windowHeight;
      this.setState({
        isVisible,
      });
    }
  }
  public render() {
    const { image, title } = this.props;
    return (
      <div ref={(ref) => { this.divRef = ref; }}>
        <WithAnimation type="fadeInUp" delay={100} {...this.state}>
          <Box pad="large">
            <Relative>
              <Overlay>
                <p>{title}</p>
              </Overlay>
              <Image imageSize="large" src={image} alt={title} />
            </Relative>
          </Box>
        </WithAnimation>
      </div>
    );
  }
}
