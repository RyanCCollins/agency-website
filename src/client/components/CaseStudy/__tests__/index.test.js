import { shallow, mount } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import React from 'react';
import { Box, Image, WithAnimation, Anchor } from 'ui';
import Overlay from '../overlay';
import Relative from '../relative';
import CaseStudy from '../index';
import props from './__mocks__/caseStudy.mock';

describe('<CaseStudy />', () => {
  describe('when shallow rendered', () => {
    const wrapper = shallow(
      <CaseStudy {...props} />,
    );
    it('should create a snapshot', () => {
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

      expect(wrapper.find(Box).length).toBe(1);
    it('should have a box component', () => {
      expect(wrapper.find(Box).length).toBe(1);
    });

    it('should have a Image component', () => {
      expect(wrapper.find(Image).length).toBe(1);
    });

    it('should have a WithAnimation component', () => {
      expect(wrapper.find(WithAnimation).length).toBe(1);
    });

    it('should have a Anchor component', () => {
      expect(wrapper.find(Anchor).length).toBe(1);
    });

    it('should have a Overlay component', () => {
      expect(wrapper.find(Overlay).length).toBe(1);
    });

    it('should have a Relative component', () => {
      expect(wrapper.find(Relative).length).toBe(1);
    });
  });

  describe('when mounted', () => {
    const wrapper = mount(
      <CaseStudy {...props} />,
    );
    
    it('should create a snapshot', () => {
      expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('should have a handleScroll method', () => {
      expect(wrapper.instance().handleScroll).toBeDefined()
    });

    it('should have a divRef defined', () => {
      expect(wrapper.instance().divRef).toBeDefined()
    });

    it('should call handle scroll on componentDidMount', () => {
      const before = wrapper;
      const mock = jest.fn();
      wrapper.instance().handleScroll = mock;
      wrapper.instance().componentDidMount();
      expect(mock.mock.calls.length).toBe(1);
    });
  });
});