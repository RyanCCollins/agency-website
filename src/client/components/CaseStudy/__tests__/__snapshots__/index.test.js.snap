// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CaseStudy /> when mounted should create a snapshot 1`] = `
<CaseStudy
  __typename="Project"
  id="58f43041f445ed0e26cd46e6"
  image="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
  title="HPE Brand Central"
>
  <withRouter(Anchor)
    path="/case-study/58f43041f445ed0e26cd46e6"
    plain={true}
  >
    <Anchor
      method="push"
      path="/case-study/58f43041f445ed0e26cd46e6"
      plain={true}
    >
      <styled.a
        onClick={[Function]}
        plain={true}
      >
        <a
          className="ckTJUm"
          onClick={[Function]}
        >
          <div>
            <WithAnimation
              duration={1000}
              isVisible={true}
            >
              <styled.div
                delay={0}
                duration={1000}
                isVisible={true}
                type="fadeIn"
              >
                <div
                  className="jOKFwR"
                  type="fadeIn"
                >
                  <Box
                    pad="large"
                  >
                    <styled.div
                      pad="large"
                    >
                      <div
                        className="bWdHyO"
                      >
                        <styled.div>
                          <div
                            className="innYbI"
                          >
                            <styled.div>
                              <div
                                className="kPGFVC"
                              >
                                <p>
                                  HPE Brand Central
                                </p>
                              </div>
                            </styled.div>
                            <styled.img
                              alt="HPE Brand Central"
                              imageSize="large"
                              src="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
                              style={
                                Object {
                                  "maxWidth": 700,
                                }
                              }
                            >
                              <img
                                alt="HPE Brand Central"
                                className="gcaPMe"
                                src="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
                                style={
                                  Object {
                                    "maxWidth": 700,
                                  }
                                }
                              />
                            </styled.img>
                          </div>
                        </styled.div>
                      </div>
                    </styled.div>
                  </Box>
                </div>
              </styled.div>
            </WithAnimation>
          </div>
        </a>
      </styled.a>
    </Anchor>
  </withRouter(Anchor)>
</CaseStudy>
`;

exports[`<CaseStudy /> when shallow rendered should create a snapshot 1`] = `
<withRouter(Anchor)
  path="/case-study/58f43041f445ed0e26cd46e6"
  plain={true}
>
  <div>
    <WithAnimation
      duration={1000}
      isVisible={false}
    >
      <Box
        pad="large"
      >
        <styled.div>
          <styled.div>
            <p>
              HPE Brand Central
            </p>
          </styled.div>
          <styled.img
            alt="HPE Brand Central"
            imageSize="large"
            src="https://github.com/RyanCCollins/cdn/blob/master/brand-central/bcmockup.jpg?raw=true"
            style={
              Object {
                "maxWidth": 700,
              }
            }
          />
        </styled.div>
      </Box>
    </WithAnimation>
  </div>
</withRouter(Anchor)>
`;
